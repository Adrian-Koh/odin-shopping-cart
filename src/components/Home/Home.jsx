import { Link } from "react-router-dom";
import styles from "./Home.module.css";

const Home = () => {
  return (
    <div className={styles.largeContainer}>
      <div className={styles.container}>
        <h1>The Everything Shop</h1>
        <p className={styles.welcome}>
          Welcome to The Everything Shop! Here, you will find anything you can
          think of. From clothes to electronics, we've got you covered. Browse
          through our catalog and you'll find that you'll never need to visit
          another store again. We provide free shipping to all countries for a
          minimum order of $100. Happy browsing!
        </p>
        <hr />
        <div className={styles.introContainer}>
          <img
            src="../../../public/old-man-portrait.jpg"
            alt="Kier Eagan"
            className={styles.introImage}
          />
          <p className={styles.imgCredits}>
            Photo by{" "}
            <a href="https://unsplash.com/@birminghammuseumstrust?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash">
              Birmingham Museums Trust
            </a>{" "}
            on{" "}
            <a href="https://unsplash.com/photos/man-in-black-suit-painting-G70apFxSF_g?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash">
              Unsplash
            </a>
          </p>
          <p className={styles.intro}>
            The Everything Shop was founded in Kingston, NY in 1865 by Kier
            Eagan. Driven by the desire to provide a one-stop-shop for the
            community's everyday needs, Kier worked with local suppliers to put
            as many essentials as he could on the shelves. As his vision grew to
            provide access to essential goods to everywhere in the world, The
            Everything Shop has expanded to other locations which includes
            Chicago, IL, Houston, TX and San Francisco, CA, among other
            locations. With the internet, we are proud to say that you can
            purchase our products from anywhere in the world. Thank you for
            stopping by!
          </p>
        </div>
      </div>
    </div>
  );
};

export { Home };
