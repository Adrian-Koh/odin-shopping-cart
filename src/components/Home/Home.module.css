.container {
    width: 600px;
    display: flex;
    flex-direction: column;
}

.largeContainer {
    display: flex;
    justify-content: center;
}

.introContainer {
    display: grid;
    grid-template-columns: 1fr 1fr;
    margin-top: 30px;
    justify-items: center;
}

.intro {
    grid-column: 2 / 3;
    grid-row: 1 / 3;
    margin: 0;
    align-self: center;
}

.introImage {
    width: 200px;
    height: auto;
}

.imgCredits {
    font-size: 10px;
    padding: 2px;
}

.imgCredits a {
    font-size: inherit;
    margin: 0;
}

hr {
    border: solid 1px black;
    width: 100%;
    margin-top: 20px;
}

.container h1, .container p {
    background-color: white;
    border-radius: 5px;
    padding: 10px;
}