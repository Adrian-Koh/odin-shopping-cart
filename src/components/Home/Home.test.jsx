import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Home } from "./Home";

describe("Home component", () => {
  it("renders home", () => {
    const { container } = render(<Home />);
    expect(container).toMatchSnapshot();
  });
});
