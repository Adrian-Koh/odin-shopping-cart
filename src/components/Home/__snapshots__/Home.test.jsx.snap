// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home component > renders home 1`] = `
<div>
  <div
    class="_largeContainer_d6eec2"
  >
    <div
      class="_container_d6eec2"
    >
      <h1>
        The Everything Shop
      </h1>
      <p
        class="_welcome_d6eec2"
      >
        Welcome to The Everything Shop! Here, you will find anything you can think of. From clothes to electronics, we've got you covered. Browse through our catalog and you'll find that you'll never need to visit another store again. We provide free shipping to all countries for a minimum order of $100. Happy browsing!
      </p>
      <hr />
      <div
        class="_introContainer_d6eec2"
      >
        <img
          alt="Kier Eagan"
          class="_introImage_d6eec2"
          src="../../../public/old-man-portrait.jpg"
        />
        <p
          class="_imgCredits_d6eec2"
        >
          Photo by
           
          <a
            href="https://unsplash.com/@birminghammuseumstrust?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash"
          >
            Birmingham Museums Trust
          </a>
           
          on
           
          <a
            href="https://unsplash.com/photos/man-in-black-suit-painting-G70apFxSF_g?utm_content=creditCopyText&utm_medium=referral&utm_source=unsplash"
          >
            Unsplash
          </a>
        </p>
        <p
          class="_intro_d6eec2"
        >
          The Everything Shop was founded in Kingston, NY in 1865 by Kier Eagan. Driven by the desire to provide a one-stop-shop for the community's everyday needs, Kier worked with local suppliers to put as many essentials as he could on the shelves. As his vision grew to provide access to essential goods to everywhere in the world, The Everything Shop has expanded to other locations which includes Chicago, IL, Houston, TX and San Francisco, CA, among other locations. With the internet, we are proud to say that you can purchase our products from anywhere in the world. Thank you for stopping by!
        </p>
      </div>
    </div>
  </div>
</div>
`;
